test/functional/tests/io_class/test_io_class_lvm_on_cas.py

```diff
@@ -0,0 +1,153 @@
+from test_tools.fio.fio_param import ReadWrite, IoEngine
+from test_utils.size import Size, Unit
+from tests.io_class.test_io_class_prevent_wrong_configuration import \
+    create_and_load_default_io_class_config
+
+mountpoint = "/mnt/"
+io_target = "/var/test"
+
+
+@pytest.mark.require_disk("cache", DiskTypeSet([DiskType.optane, DiskType.nand]))
+@pytest.mark.require_disk("core", DiskTypeLowerThan("cache"))
+def test_io_class_lvm_on_cas():
+    """
+        title: IO class for CAS device behind LVM.
+        description: Validate the ability of CAS to cache IO class when CAS device is used as LVM.
+        pass_criteria:
+          - Create CAS device and LVM on top of it successfully.
+          - Loading IO class configuration successfully.
+          - Running FIO for file size from IO class from 11 to 21 successfully.
+          - Increasing proper statistics as expected.
+    """
+
+    with TestRun.step(f"Create CAS device."):
+        cache, core = prepare_devices(TestRun.disks['cache'], TestRun.disks['core'],
+                                      Size(20, Unit.GibiByte), CacheMode.WB)
+
+    with TestRun.step("Create LVM on CAS device."):
+        lvm_filters = ["a/.*/", "r|/dev/sd*|", "r|/dev/hd*|", "r|/dev/xvd*|", "r/disk/", "r/block/",
+                       "r|/dev/nvme*|"]
+        pv_num = 1
+        vg_num = 1
+        lv_num = 1
+        cache_num = 1
+        cas_dev_num = 1
+
+        config = LvmConfiguration(lvm_filters, pv_num, vg_num, lv_num, cache_num, cas_dev_num)
+
+        lvms = Lvm.create_specific_lvm_configuration(core, config)
+        lvm = lvms[0]
+
+    with TestRun.step("Create filesystem for LVM and mount it."):
+        lvm.create_filesystem(Filesystem.ext4)
+        lvm.mount(mountpoint)
+
+    with TestRun.step("Load IO class config."):
+        create_and_load_default_io_class_config(cache)
+
+    with TestRun.step("Run fio for file size from IO class from 11 to 21 "
+                      "and check that correct statistics increase."):
+        csv = casadm.list_io_classes(cache.cache_id, OutputFormat.csv).stdout
+        io_classes = IoClass.csv_to_list(csv)
+        file_size = Size(2, Unit.KibiByte)
+
+        for io_class in io_classes:
+            if io_class.id < 11 or io_class.id > 21:
+                continue
+
+            TestRun.LOGGER.info(f"IO Class ID: {io_class.id}, class name: {io_class.rule}")
+            cache.reset_counters()
+
+            TestRun.LOGGER.info(f"Running FIO with verification on LVM "
+                                f"[IO class ID {io_class.id}].")
+            run_fio(file_size, io_target)
+
+            TestRun.LOGGER.info(f"Checking statistics [IO class ID {io_class.id}].")
+            check_statistics(cache, io_classes, io_class.id)
+
+            if file_size < Size(256, Unit.MebiByte):
+                file_size *= 4
+            else:
+                file_size = 1100 * Size(1, Unit.MebiByte)
+
+            fs_utils.remove(io_target)
```

Comment:
Maybe I don't understand something. but I don't see how CAS device is actually used here.